src/group_theory/free_product.lean

```diff
@@ -327,4 +328,19 @@ instance : decidable_eq (free_product M) := word.equiv.decidable_eq
 
 end word
 
+/-- The free product of free groups is itself a free group -/
+@[simps]
+instance {ι : Type*} (G : ι → Type*) [∀ i, group (G i)] [hG : ∀ i, is_free_group (G i)] :
+  is_free_group (free_product G) :=
+{ generators := Σ i, is_free_group.generators (G i),
+  of := λ x, free_product.of (is_free_group.of x.2),
+  unique_lift' :=
+  begin
+    introsI X _ f,
+    refine ⟨free_product.lift (λ i, is_free_group.lift (λ x, f ⟨i, x⟩)), _ ⟩,
+    split,
+    { simp, },
+    { intros g hfg, ext i x, simpa using hfg ⟨i, x⟩, }
```

Comment:
You don't need to follow a refine like this with `split`:
```suggestion
    refine ⟨free_product.lift (λ i, is_free_group.lift (λ x, f ⟨i, x⟩)), by simp, λ g hfg, _⟩,
    ext i x,
    simpa using hfg ⟨i, x⟩,
```
(+ bonus tweakz)

Reply:
Ups, didn't see this comment before I fired of bors, sorry. Ok to leave the bors batch running?

Reply:
Not worth interrupting bors for something so small.